node/node.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package node
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/extensions/adminactions"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/servers"
+	"github.com/rackspace/gophercloud/pagination"
+	"github.com/stackanetes/kubernetes-entrypoint/util/command"
+	"github.com/stackanetes/kubernetes-entrypoint/logger"
+	"time"
+)
+
+const (
+	interval = 2
+)
+
+// Node is the implementation of a openstack hypervisor.\
+type Node struct {
+	client   *gophercloud.ServiceClient
+	body     map[string]string
+	confPath string
+	hostname string
+	vms      *[]servers.Server
+	Enabled  bool
+}
+
+// New creates a Openstack node.
+func New(confPath string) (*Node, error) {
+	var n Node
+	var err error
+
+	n.Enabled = true
+	n.confPath = confPath
+	n.hostname, err = GetMyHostname()
+	if err != nil {
+		return &n, fmt.Errorf("Cannot retrieve hostname: %v", err)
+	}
+
+	n.body = map[string]string{"binary": "nova-compute", "host": n.hostname}
+
+	n.client, err = createOpenstackClient(n.confPath)
```

Comment:
We need a retry for creating a client (at least three tries, with timeout beetwen them).